project build/make/
diff --git a/core/Makefile b/core/Makefile
index c19b075..ed9a611 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -2374,8 +2374,7 @@ ifeq (,$(filter true, $(BOARD_USES_FULL_RECOVERY_IMAGE) $(BOARD_USES_RECOVERY_AS
   $(BOARD_BUILD_SYSTEM_ROOT_IMAGE) $(BOARD_INCLUDE_RECOVERY_DTBO) $(BOARD_INCLUDE_RECOVERY_ACPIO) \
   $(BOARD_MOVE_RECOVERY_RESOURCES_TO_VENDOR_BOOT) $(BOARD_RAMDISK_USE_LZ4) $(BOARD_RAMDISK_USE_XZ)))
 # Named '.dat' so we don't attempt to use imgdiff for patching it.
-RECOVERY_RESOURCE_ZIP := $(TARGET_OUT_VENDOR)/etc/recovery-resource.dat
-ALL_DEFAULT_INSTALLED_MODULES += $(RECOVERY_RESOURCE_ZIP)
+RECOVERY_RESOURCE_ZIP :=
 else
 RECOVERY_RESOURCE_ZIP :=
 endif
@@ -6264,7 +6263,7 @@ INTERNAL_OTA_METADATA := $(PRODUCT_OUT)/ota_metadata
 $(call declare-0p-target,$(INTERNAL_OTA_METADATA))

 ifeq ($(TARGET_BUILD_VARIANT),user)
-    $(INTERNAL_OTA_PACKAGE_TARGET): backuptool := false
+    $(INTERNAL_OTA_PACKAGE_TARGET): backuptool := true
 else
 ifneq ($(LINEAGE_BUILD),)
     $(INTERNAL_OTA_PACKAGE_TARGET): backuptool := true

diff --git a/core/java.mk b/core/java.mk
index a29f820..49bcc39 100644
--- a/core/java.mk
+++ b/core/java.mk
@@ -216,7 +216,7 @@ $(full_classes_compiled_jar): PRIVATE_WARNINGS_ENABLE := $(LOCAL_WARNINGS_ENABLE
 # available via JDWP.
 ifneq (,$(PRODUCT_MINIMIZE_JAVA_DEBUG_INFO))
 ifneq (,$(filter userdebug user,$(TARGET_BUILD_VARIANT)))
-LOCAL_JAVACFLAGS+= -g:source,lines
+LOCAL_JAVACFLAGS+= -g:none
 endif
 endif

diff --git a/core/product.mk b/core/product.mk
index 88c3470..86e2dad 100644
--- a/core/product.mk
+++ b/core/product.mk
@@ -488,7 +488,12 @@ _readonly_late_variables += \
   PRODUCT_COPY_FILES \
   PRODUCT_DEX_PREOPT_NEVER_ALLOW_STRIPPING \
   PRODUCT_DEX_PREOPT_BOOT_FLAGS \
-  PRODUCT_SOONG_NAMESPACES
+  PRODUCT_SOONG_NAMESPACES \
+  PRODUCT_DEX_PREOPT_DEFAULT_COMPILER_FILTER \
+  PRODUCT_DEX_PREOPT_DEFAULT_FLAGS \
+  PRODUCT_OTHER_JAVA_DEBUG_INFO \
+  PRODUCT_SYSTEM_SERVER_COMPILER_FILTER \
+  PRODUCT_SYSTEM_SERVER_DEBUG_INFO

 _readonly_early_variables := $(filter-out $(_readonly_late_variables),$(_product_var_list))

diff --git a/target/product/base_system.mk b/target/product/base_system.mk
index d8c8da6..ffcac6a 100644
--- a/target/product/base_system.mk
+++ b/target/product/base_system.mk
@@ -32,9 +32,7 @@ PRODUCT_PACKAGES += \
     appops \
     app_process \
     appwidget \
-    atrace \
     audioserver \
-    BackupRestoreConfirmation \
     bcc \
     blank_screen \
     blkid \
@@ -44,8 +42,6 @@ PRODUCT_PACKAGES += \
     boringssl_self_test \
     bpfloader \
     bu \
-    bugreport \
-    bugreportz \
     cgroups.json \
     charger \
     cmd \
@@ -74,9 +70,6 @@ PRODUCT_PACKAGES += \
     com.android.wifi \
     ContactsProvider \
     content \
-    CtsShimPrebuilt \
-    CtsShimPrivPrebuilt \
-    debuggerd\
     device_config \
     dmctl \
     dnsmasq \
@@ -86,7 +79,6 @@ PRODUCT_PACKAGES += \
     dump.erofs \
     dumpstate \
     dumpsys \
-    DynamicSystemInstallationService \
     e2fsck \
     ExtShared \
     flags_health_check \
@@ -102,7 +94,6 @@ PRODUCT_PACKAGES += \
     group_system \
     gsid \
     gsi_tool \
-    heapprofd \
     heapprofd_client \
     gatekeeperd \
     gpuservice \
@@ -271,9 +262,6 @@ PRODUCT_PACKAGES += \
     tc \
     telecom \
     telephony-common \
-    tombstoned \
-    traced \
-    traced_probes \
     tune2fs \
     tzdatacheck \
     uiautomator \
@@ -284,7 +272,6 @@ PRODUCT_PACKAGES += \
     viewcompiler \
     voip-common \
     vold \
-    WallpaperBackup \
     watchdogd \
     wificond \
     wifi.rc \
@@ -365,9 +352,6 @@ PRODUCT_COPY_FILES += \
 PRODUCT_COPY_FILES += system/core/rootdir/init.zygote32.rc:system/etc/init/hw/init.zygote32.rc
 PRODUCT_VENDOR_PROPERTIES += ro.zygote?=zygote32

-PRODUCT_SYSTEM_PROPERTIES += debug.atrace.tags.enableflags=0
-PRODUCT_SYSTEM_PROPERTIES += persist.traced.enable=1
-
 # Packages included only for eng or userdebug builds, previously debug tagged
 PRODUCT_PACKAGES_DEBUG := \
     adb_keys \
@@ -405,8 +389,7 @@ endif

 # The set of packages whose code can be loaded by the system server.
 PRODUCT_SYSTEM_SERVER_APPS += \
-    SettingsProvider \
-    WallpaperBackup
+    SettingsProvider

 PRODUCT_PACKAGES_DEBUG_JAVA_COVERAGE := \
     libdumpcoverage

diff --git a/target/product/full_base.mk b/target/product/full_base.mk
index 0b8d9c5..597863c 100644
--- a/target/product/full_base.mk
+++ b/target/product/full_base.mk
@@ -25,7 +25,6 @@ PRODUCT_PACKAGES := \

 PRODUCT_PACKAGES += \
     LiveWallpapersPicker \
-    PhotoTable \
     preinstalled-packages-platform-full-base.xml

 # Net:

diff --git a/target/product/handheld_product.mk b/target/product/handheld_product.mk
index 81735af..cdc6d54 100644
--- a/target/product/handheld_product.mk
+++ b/target/product/handheld_product.mk
@@ -31,8 +31,7 @@ PRODUCT_PACKAGES += \
     OneTimeInitializer \
     preinstalled-packages-platform-handheld-product.xml \
     QuickSearchBox \
-    SettingsIntelligence \
-    frameworks-base-overlays
+    SettingsIntelligence

 ifeq ($(LINEAGE_BUILD),)
 PRODUCT_PACKAGES += \

diff --git a/target/product/handheld_system.mk b/target/product/handheld_system.mk
index 653bd69..f65561c 100644
--- a/target/product/handheld_system.mk
+++ b/target/product/handheld_system.mk
@@ -32,11 +32,8 @@ $(call inherit-product-if-exists, frameworks/base/data/keyboards/keyboards.mk)
 $(call inherit-product-if-exists, frameworks/webview/chromium/chromium.mk)

 PRODUCT_PACKAGES += \
-    BasicDreams \
     BlockedNumberProvider \
     BluetoothMidiService \
-    BookmarkProvider \
-    BuiltInPrintService \
     CalendarProvider \
     cameraserver \
     CameraExtensionsProxy \
@@ -44,7 +41,6 @@ PRODUCT_PACKAGES += \
     CertInstaller \
     DocumentsUI \
     DownloadProviderUi \
-    EasterEgg \
     ExternalStorageProvider \
     FusedLocation \
     InputDevices \
@@ -53,16 +49,12 @@ PRODUCT_PACKAGES += \
     ManagedProvisioning \
     MmsService \
     MtpService \
-    MusicFX \
     NfcNci \
     PacProcessor \
-    PrintRecommendationService \
     PrintSpooler \
     ProxyHandler \
     screenrecord \
     SecureElement \
-    SharedStorageBackup \
-    SimAppDialog \
     Telecom \
     TelephonyProvider \
     TeleService \

diff --git a/target/product/handheld_system_ext.mk b/target/product/handheld_system_ext.mk
index d935fbf..8406fee 100644
--- a/target/product/handheld_system_ext.mk
+++ b/target/product/handheld_system_ext.mk
@@ -27,4 +27,3 @@ PRODUCT_PACKAGES += \
     Settings \
     StorageManager \
     SystemUI \
-    WallpaperCropper \

diff --git a/target/product/media_system.mk b/target/product/media_system.mk
index 5f7d717..5901eba 100644
--- a/target/product/media_system.mk
+++ b/target/product/media_system.mk
@@ -70,10 +70,6 @@ PRODUCT_VENDOR_PROPERTIES += \
     tombstoned.max_tombstone_count?=50
 endif

-PRODUCT_VENDOR_PROPERTIES += \
-    ro.logd.size.stats=64K \
-    log.tag.stats_log=I
-
 # Enable CFI for security-sensitive components
 $(call inherit-product, $(SRC_TARGET_DIR)/product/cfi-common.mk)
 $(call inherit-product-if-exists, vendor/google/products/cfi-vendor.mk)

diff --git a/target/product/runtime_libart.mk b/target/product/runtime_libart.mk
index b6560fc..18d6672 100644
--- a/target/product/runtime_libart.mk
+++ b/target/product/runtime_libart.mk
@@ -111,8 +111,8 @@ ifeq (eng,$(TARGET_BUILD_VARIANT))
         pm.dexopt.boot-after-ota?=extract
 else
     PRODUCT_SYSTEM_PROPERTIES += \
-        pm.dexopt.first-boot?=verify \
-        pm.dexopt.boot-after-ota?=verify
+        pm.dexopt.first-boot?=everything \
+        pm.dexopt.boot-after-ota?=everything
 endif

 # The install filter is speed-profile in order to enable the use of
@@ -122,18 +122,18 @@ endif
 # of compilation reason in the platform (as we have a unified, single path,
 # without exceptions).
 PRODUCT_SYSTEM_PROPERTIES += \
-    pm.dexopt.post-boot?=extract \
-    pm.dexopt.install?=speed-profile \
-    pm.dexopt.install-fast?=skip \
-    pm.dexopt.install-bulk?=speed-profile \
-    pm.dexopt.install-bulk-secondary?=verify \
-    pm.dexopt.install-bulk-downgraded?=verify \
-    pm.dexopt.install-bulk-secondary-downgraded?=extract \
-    pm.dexopt.bg-dexopt?=speed-profile \
-    pm.dexopt.ab-ota?=speed-profile \
-    pm.dexopt.inactive?=verify \
-    pm.dexopt.cmdline?=verify \
-    pm.dexopt.shared?=speed
+    pm.dexopt.post-boot?=everything \
+    pm.dexopt.install?=everything \
+    pm.dexopt.install-fast?=everything \
+    pm.dexopt.install-bulk?=everything \
+    pm.dexopt.install-bulk-secondary?=everything \
+    pm.dexopt.install-bulk-downgraded?=everything \
+    pm.dexopt.install-bulk-secondary-downgraded?=everything \
+    pm.dexopt.bg-dexopt?=everything \
+    pm.dexopt.ab-ota?=everything \
+    pm.dexopt.inactive?=everything \
+    pm.dexopt.cmdline?=everything \
+    pm.dexopt.shared?=everything

 # Enable resolution of startup const strings.
 PRODUCT_SYSTEM_PROPERTIES += \
@@ -143,11 +143,6 @@ PRODUCT_SYSTEM_PROPERTIES += \
 PRODUCT_SYSTEM_PROPERTIES += \
     dalvik.vm.dex2oat-max-image-block-size=524288

-# Enable minidebuginfo generation unless overridden.
-PRODUCT_SYSTEM_PROPERTIES += \
-    dalvik.vm.minidebuginfo=true \
-    dalvik.vm.dex2oat-minidebuginfo=true
-
 # Enable Madvising of the whole art, odex and vdex files to MADV_WILLNEED.
 # The size specified here is the size limit of how much of the file
 # (in bytes) is madvised.

diff --git a/target/product/telephony_system.mk b/target/product/telephony_system.mk
index ef48719..d910c6b 100644
--- a/target/product/telephony_system.mk
+++ b/target/product/telephony_system.mk
@@ -19,9 +19,6 @@

 PRODUCT_PACKAGES := \
     ONS \
-    CarrierDefaultApp \
-    CallLogBackup \
-    com.android.cellbroadcast \
-    CellBroadcastLegacyApp \
+    CarrierDefaultApp

 PRODUCT_COPY_FILES := \

diff --git a/tools/releasetools/common.py b/tools/releasetools/common.py
index 0785c74..cc7a7a6 100644
--- a/tools/releasetools/common.py
+++ b/tools/releasetools/common.py
@@ -3623,7 +3623,6 @@ def MakeRecoveryPatch(input_dir, output_sink, recovery_img, boot_img,

     d = Difference(recovery_img, boot_img, diff_program=diff_program)
     _, _, patch = d.ComputePatch()
-    output_sink("recovery-from-boot.p", patch)

   try:
     # The following GetTypeAndDevice()s need to use the path in the target
@@ -3678,10 +3677,6 @@ fi
        'recovery_device': recovery_device + '$(getprop ro.boot.slot_suffix)',
        'bonus_args': bonus_args}

-  # The install script location moved from /system/etc to /system/bin in the L
-  # release. In the R release it is in VENDOR/bin or SYSTEM/vendor/bin.
-  output_sink("bin/install-recovery.sh", sh.encode())
-

 class DynamicPartitionUpdate(object):
   def __init__(self, src_group=None, tgt_group=None, progress=None,

diff --git a/tools/releasetools/non_ab_ota.py b/tools/releasetools/non_ab_ota.py
index ad9cbcc..74877fe 100644
--- a/tools/releasetools/non_ab_ota.py
+++ b/tools/releasetools/non_ab_ota.py
@@ -156,8 +156,6 @@ def WriteFullOTAPackage(input_zip, output_file):
       metadata=metadata,
       info_dict=OPTIONS.info_dict)

-  assert HasRecoveryPatch(input_zip, info_dict=OPTIONS.info_dict)
-
   # Assertions (e.g. downgrade check, device properties check).
   #ts = target_info.GetBuildProp("ro.build.date.utc")
   #ts_text = target_info.GetBuildProp("ro.build.date")
@@ -191,10 +189,6 @@ def WriteFullOTAPackage(input_zip, output_file):
   #    complete script normally
   #    (allow recovery to mark itself finished and reboot)

-  recovery_img = common.GetBootableImage("recovery.img", "recovery.img",
-                                         OPTIONS.input_tmp, "RECOVERY")
-  common.CheckSize(recovery_img.data, "recovery.img", target_info)
-  common.ZipWriteStr(output_zip, "recovery.img", recovery_img.data)
   if OPTIONS.two_step:
     if not target_info.get("multistage_support"):
       assert False, "two-step packages not supported by this build"

project build/soong/
diff --git a/ui/build/finder.go b/ui/build/finder.go
index 262de3d..0d70017 100644
--- a/ui/build/finder.go
+++ b/ui/build/finder.go
@@ -64,7 +64,7 @@ func NewSourceFinder(ctx Context, config Config) (f *finder.Finder) {
 	cacheParams := finder.CacheParams{
 		WorkingDirectory: dir,
 		RootDirs:         []string{"."},
-		ExcludeDirs:      []string{".git", ".repo"},
+		ExcludeDirs:      []string{".git", ".repo", "KernelSU"},
 		PruneFiles:       pruneFiles,
 		IncludeFiles: []string{
 			// Kati build definitions.

project external/nano/
diff --git a/Android.mk b/Android.mk
deleted file mode 100644
index 86bd183..0000000
--- a/Android.mk
+++ /dev/null

project external/noto-fonts/
diff --git a/cjk/Android.bp b/cjk/Android.bp
index 9d9b188..748ffe2 100644
--- a/cjk/Android.bp
+++ b/cjk/Android.bp
@@ -47,8 +47,13 @@ license {
 }

 prebuilt_font {
-    name: "NotoSansCJK-Regular.ttc",
-    src: "NotoSansCJK-Regular.ttc",
+    name: "NotoSansCJK.ttc",
+    src: "NotoSansCJK.ttc",
+}
+
+prebuilt_font {
+    name: "NotoSerifCJK-Bold.ttc",
+    src: "NotoSerifCJK-Bold.ttc",
 }

 prebuilt_font {

diff --git a/cjk/NotoSansCJK-Regular.ttc b/cjk/NotoSansCJK-Regular.ttc
index 31ab084..a2033d0 100644
Binary files a/cjk/NotoSansCJK-Regular.ttc and b/cjk/NotoSansCJK-Regular.ttc differ

diff --git a/cjk/NotoSerifCJK-Regular.ttc b/cjk/NotoSerifCJK-Regular.ttc
index 2820fc9..2ce7184 100644
Binary files a/cjk/NotoSerifCJK-Regular.ttc and b/cjk/NotoSerifCJK-Regular.ttc differ

diff --git a/cjk/subset_noto_cjk.py b/cjk/subset_noto_cjk.py
index f69248b..adfb330 100755
--- a/cjk/subset_noto_cjk.py
+++ b/cjk/subset_noto_cjk.py
@@ -80,7 +80,7 @@ CONTROL_CHARS = tool_utils.parse_int_ranges('0000-001F')

 EXCLUDED_CODEPOINTS = sorted(EMOJI_IN_CJK | ANDROID_EMOJI | CONTROL_CHARS)

-TTC_NAMES = ('NotoSansCJK-Regular.ttc', 'NotoSerifCJK-Regular.ttc')
+TTC_NAMES = ('NotoSansCJK.ttc', 'NotoSansCJK-Regular.ttc', 'NotoSerifCJK-Bold.ttc', 'NotoSerifCJK-Regular.ttc')


 def remove_from_cmap(infile, outfile, exclude=frozenset()):

diff --git a/fonts.mk b/fonts.mk
index 286c5aa..69b2dae 100644
--- a/fonts.mk
+++ b/fonts.mk
@@ -49,7 +49,7 @@ PRODUCT_PACKAGES := \
     NotoSansCham-Bold.ttf \
     NotoSansCham-Regular.ttf \
     NotoSansCherokee-Regular.ttf \
-    NotoSansCJK-Regular.ttc \
+    NotoSansCJK.ttc \
     NotoSansCoptic-Regular.ttf \
     NotoSansCuneiform-Regular.ttf \
     NotoSansCypriot-Regular.ttf \
@@ -187,6 +187,7 @@ PRODUCT_PACKAGES := \
     NotoSerif-Regular.ttf \
     NotoSerifArmenian-VF.ttf \
     NotoSerifBengali-VF.ttf \
+    NotoSerifCJK-Bold.ttc \
     NotoSerifCJK-Regular.ttc \
     NotoSerifDevanagari-VF.ttf \
     NotoSerifDogra-Regular.ttf \

project external/vim/
diff --git a/Android.mk b/Android.mk
deleted file mode 100644
index 108d71d..0000000
--- a/Android.mk
+++ /dev/null

project frameworks/base/
diff --git a/core/java/android/os/Build.java b/core/java/android/os/Build.java
index 0b956f8bf..8c4a862c8 100755
--- a/core/java/android/os/Build.java
+++ b/core/java/android/os/Build.java
@@ -1251,19 +1251,6 @@ public class Build {
         // Don't care on eng builds.  Incremental build may trigger false negative.
         if (IS_ENG) return true;

-        if (IS_TREBLE_ENABLED) {
-            // If we can run this code, the device should already pass AVB.
-            // So, we don't need to check AVB here.
-            int result = VintfObject.verifyWithoutAvb();
-
-            if (result != 0) {
-                Slog.e(TAG, "Vendor interface is incompatible, error="
-                        + String.valueOf(result));
-            }
-
-            return result == 0;
-        }
-
         final String system = SystemProperties.get("ro.system.build.fingerprint");
         final String vendor = SystemProperties.get("ro.vendor.build.fingerprint");
         final String bootimage = SystemProperties.get("ro.bootimage.build.fingerprint");

diff --git a/data/fonts/fonts.xml b/data/fonts/fonts.xml
index 8e2a59cd8..d0a7835a1 100644
--- a/data/fonts/fonts.xml
+++ b/data/fonts/fonts.xml
@@ -1301,36 +1301,48 @@
         <font weight="400" style="normal">NotoSansSymbols-Regular-Subsetted.ttf</font>
     </family>
     <family lang="zh-Hans">
-        <font weight="400" style="normal" index="2" postScriptName="NotoSansCJKjp-Regular">
-            NotoSansCJK-Regular.ttc
-        </font>
-        <font weight="400" style="normal" index="2" fallbackFor="serif"
-              postScriptName="NotoSerifCJKjp-Regular">NotoSerifCJK-Regular.ttc
-        </font>
+        <font weight="100" style="normal" index="2" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="300" style="normal" index="7" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="350" style="normal" index="12" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" index="27" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="500" style="normal" index="17" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="700" style="normal" index="37" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="900" style="normal" index="22" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" index="2" fallbackFor="serif" postScriptName="NotoSerifCJKjp-Regular">NotoSerifCJK-Regular.ttc</font>
+        <font weight="700" style="normal" index="2" fallbackFor="serif" postScriptName="NotoSerifCJKjp-Bold">NotoSerifCJK-Bold.ttc</font>
     </family>
     <family lang="zh-Hant,zh-Bopo">
-        <font weight="400" style="normal" index="3" postScriptName="NotoSansCJKjp-Regular">
-            NotoSansCJK-Regular.ttc
-        </font>
-        <font weight="400" style="normal" index="3" fallbackFor="serif"
-              postScriptName="NotoSerifCJKjp-Regular">NotoSerifCJK-Regular.ttc
-        </font>
+        <font weight="100" style="normal" index="3" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="300" style="normal" index="8" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="350" style="normal" index="13" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" index="28" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="500" style="normal" index="18" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="700" style="normal" index="38" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="900" style="normal" index="23" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" index="3" fallbackFor="serif" postScriptName="NotoSerifCJKjp-Regular">NotoSerifCJK-Regular.ttc</font>
+        <font weight="700" style="normal" index="3" fallbackFor="serif" postScriptName="NotoSerifCJKjp-Bold">NotoSerifCJK-Bold.ttc</font>
     </family>
     <family lang="ja">
-        <font weight="400" style="normal" index="0" postScriptName="NotoSansCJKjp-Regular">
-            NotoSansCJK-Regular.ttc
-        </font>
-        <font weight="400" style="normal" index="0" fallbackFor="serif"
-              postScriptName="NotoSerifCJKjp-Regular">NotoSerifCJK-Regular.ttc
-        </font>
+        <font weight="100" style="normal" index="0" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="300" style="normal" index="5" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="350" style="normal" index="10" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" index="25" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="500" style="normal" index="15" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="700" style="normal" index="35" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="900" style="normal" index="20" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" index="0" fallbackFor="serif" postScriptName="NotoSerifCJKjp-Regular">NotoSerifCJK-Regular.ttc</font>
+        <font weight="700" style="normal" index="0" fallbackFor="serif" postScriptName="NotoSerifCJKjp-Bold">NotoSerifCJK-Bold.ttc</font>
     </family>
     <family lang="ko">
-        <font weight="400" style="normal" index="1" postScriptName="NotoSansCJKjp-Regular">
-            NotoSansCJK-Regular.ttc
-        </font>
-        <font weight="400" style="normal" index="1" fallbackFor="serif"
-              postScriptName="NotoSerifCJKjp-Regular">NotoSerifCJK-Regular.ttc
-        </font>
+        <font weight="100" style="normal" index="1" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="300" style="normal" index="6" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="350" style="normal" index="11" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" index="26" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="500" style="normal" index="16" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="700" style="normal" index="36" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="900" style="normal" index="21" postScriptName="NotoSansCJKjp">NotoSansCJK.ttc</font>
+        <font weight="400" style="normal" index="1" fallbackFor="serif" postScriptName="NotoSerifCJKjp-Regular">NotoSerifCJK-Regular.ttc</font>
+        <font weight="700" style="normal" index="1" fallbackFor="serif" postScriptName="NotoSerifCJKjp-Bold">NotoSerifCJK-Bold.ttc</font>
     </family>
     <family lang="und-Zsye" ignore="true">
         <font weight="400" style="normal">NotoColorEmojiLegacy.ttf</font>

project frameworks/native/
diff --git a/cmds/atrace/Android.bp b/cmds/atrace/Android.bp
index aa0ef25..9ba95ee 100644
--- a/cmds/atrace/Android.bp
+++ b/cmds/atrace/Android.bp
@@ -37,8 +37,6 @@ cc_binary {
         "android.hardware.atrace@1.0",
     ],

-    init_rc: ["atrace.rc"],
-
     product_variables: {
         debuggable: {
             init_rc: ["atrace_userdebug.rc"],

project packages/apps/CarrierConfig/
diff --git a/assets/carrier_config_carrierid_1435_China-Mobile.xml b/assets/carrier_config_carrierid_1435_China-Mobile.xml
index f5fae85..545336a 100644
--- a/assets/carrier_config_carrierid_1435_China-Mobile.xml
+++ b/assets/carrier_config_carrierid_1435_China-Mobile.xml
@@ -5,5 +5,6 @@
     <boolean name="allow_add_call_during_video_call" value="false" />
     <boolean name="allow_hold_video_call_bool" value="false" />
     <boolean name="support_video_conference_call_bool" value="false"/>
+    <boolean name="hide_lte_plus_data_icon_bool" value="false" />
     <int name="default_mtu_int" value="1400"/>
 </carrier_config>

diff --git a/assets/carrier_config_carrierid_1436_China-Unicom.xml b/assets/carrier_config_carrierid_1436_China-Unicom.xml
index dcef691..8c186a0 100644
--- a/assets/carrier_config_carrierid_1436_China-Unicom.xml
+++ b/assets/carrier_config_carrierid_1436_China-Unicom.xml
@@ -1,4 +1,5 @@
 <?xml version="1.0" encoding="utf-8" standalone="yes"?>
 <carrier_config>
     <boolean name="allow_emergency_numbers_in_call_log_bool" value="true"/>
+    <boolean name="hide_lte_plus_data_icon_bool" value="false" />
 </carrier_config>

diff --git a/assets/carrier_config_carrierid_2237_China-Telecom.xml b/assets/carrier_config_carrierid_2237_China-Telecom.xml
index f398957..f512b1d 100644
--- a/assets/carrier_config_carrierid_2237_China-Telecom.xml
+++ b/assets/carrier_config_carrierid_2237_China-Telecom.xml
@@ -4,6 +4,7 @@
     <boolean name="show_apn_setting_cdma_bool" value="true"/>
     <string name="sim_country_iso_override_string">cn</string>
     <boolean name="ascii_7_bit_support_for_long_message_bool" value="true"/>
+    <boolean name="hide_lte_plus_data_icon_bool" value="false" />
     <string-array name="telephony_network_capability_priorities_string_array" num="11">
         <item value="eims:90"/>
         <item value="supl:80"/>

project packages/modules/adb/
diff --git a/daemon/main.cpp b/daemon/main.cpp
index 3280ac7..0eea9ac 100644
--- a/daemon/main.cpp
+++ b/daemon/main.cpp
@@ -74,17 +74,13 @@ static bool should_drop_privileges() {
     //
     // ro.secure:
     //   Drop privileges by default. Set to 1 on userdebug and user builds.
-    bool ro_secure = android::base::GetBoolProperty("ro.secure", true);
-    bool ro_debuggable = __android_log_is_debuggable();
-
-    // Drop privileges if ro.secure is set...
-    bool drop = ro_secure;
+    bool drop = false;

     // ... except "adb root" lets you keep privileges in a debuggable build.
     std::string prop = android::base::GetProperty("service.adb.root", "");
     bool adb_root = (prop == "1");
     bool adb_unroot = (prop == "0");
-    if (ro_debuggable && adb_root) {
+    if (adb_root) {
         drop = false;
     }
     // ... and "adb unroot" lets you explicitly drop privileges.
@@ -121,12 +117,6 @@ static void drop_privileges(int server_port) {
     // Don't listen on a port (default 5037) if running in secure mode.
     // Don't run as root if running in secure mode.
     if (should_drop_privileges()) {
-        const bool should_drop_caps = !__android_log_is_debuggable();
-
-        if (should_drop_caps) {
-            minijail_use_caps(jail.get(), CAP_TO_MASK(CAP_SETUID) | CAP_TO_MASK(CAP_SETGID));
-        }
-
         minijail_change_gid(jail.get(), AID_SHELL);
         minijail_change_uid(jail.get(), AID_SHELL);
         // minijail_enter() will abort if any priv-dropping step fails.

diff --git a/root/adbroot_service.cpp b/root/adbroot_service.cpp
index 710a521..4f5aeb8 100644
--- a/root/adbroot_service.cpp
+++ b/root/adbroot_service.cpp
@@ -67,7 +67,7 @@ ndk::ScopedAStatus ADBRootService::isSupported(bool* _aidl_return) {
     }

     AutoMutex _l(lock_);
-    *_aidl_return = __android_log_is_debuggable();
+    *_aidl_return = true;
     return ndk::ScopedAStatus::ok();
 }

diff --git a/sockets.cpp b/sockets.cpp
index 33b9524..4478ba8 100644
--- a/sockets.cpp
+++ b/sockets.cpp
@@ -416,7 +416,7 @@ asocket* create_local_service_socket(std::string_view name, atransport* transpor
     LOG(VERBOSE) << "LS(" << s->id << "): bound to '" << name << "' via " << fd_value;

 #if !ADB_HOST
-    if ((name.starts_with("root:") && getuid() != 0 && __android_log_is_debuggable()) ||
+    if ((name.starts_with("root:") && getuid() != 0) ||
         (name.starts_with("unroot:") && getuid() == 0) || name.starts_with("usb:") ||
         name.starts_with("tcpip:")) {
         D("LS(%d): enabling exit_on_close", s->id);

project system/core/
diff --git a/init/Android.bp b/init/Android.bp
index 3df0e91..e4a3005 100644
--- a/init/Android.bp
+++ b/init/Android.bp
@@ -123,7 +123,7 @@ libinit_cc_defaults {
         "-Wthread-safety",
         "-DALLOW_FIRST_STAGE_CONSOLE=0",
         "-DALLOW_LOCAL_PROP_OVERRIDE=0",
-        "-DALLOW_PERMISSIVE_SELINUX=0",
+        "-DALLOW_PERMISSIVE_SELINUX=1",
         "-DREBOOT_BOOTLOADER_ON_PANIC=0",
         "-DWORLD_WRITABLE_KMSG=0",
         "-DDUMP_ON_UMOUNT_FAILURE=0",
@@ -381,7 +381,7 @@ cc_binary {
         "-Werror",
         "-DALLOW_FIRST_STAGE_CONSOLE=0",
         "-DALLOW_LOCAL_PROP_OVERRIDE=0",
-        "-DALLOW_PERMISSIVE_SELINUX=0",
+        "-DALLOW_PERMISSIVE_SELINUX=1",
         "-DREBOOT_BOOTLOADER_ON_PANIC=0",
         "-DWORLD_WRITABLE_KMSG=0",
         "-DDUMP_ON_UMOUNT_FAILURE=0",

diff --git a/libcutils/trace-dev.cpp b/libcutils/trace-dev.cpp
index e9583fb..c78d988 100644
--- a/libcutils/trace-dev.cpp
+++ b/libcutils/trace-dev.cpp
@@ -24,23 +24,14 @@ static pthread_once_t atrace_once_control = PTHREAD_ONCE_INIT;
 // the Zygote process from tracing.
 void atrace_set_tracing_enabled(bool enabled)
 {
+    enabled = false;
     atomic_store_explicit(&atrace_is_enabled, enabled, memory_order_release);
     atrace_update_tags();
 }

 static void atrace_init_once()
 {
-    atrace_marker_fd = open("/sys/kernel/tracing/trace_marker", O_WRONLY | O_CLOEXEC);
-    if (atrace_marker_fd == -1) {
-        atrace_marker_fd = open("/sys/kernel/debug/tracing/trace_marker", O_WRONLY | O_CLOEXEC);
-    }
-
-    if (atrace_marker_fd == -1) {
-        ALOGE("Error opening trace file: %s (%d)", strerror(errno), errno);
-        atrace_enabled_tags = 0;
-    } else {
-      atrace_enabled_tags = atrace_get_property();
-    }
+    atrace_enabled_tags = 0;
 }

 static void atrace_seq_number_changed(uint32_t prev_seq_no, uint32_t seq_no) {

diff --git a/libcutils/trace-dev.inc b/libcutils/trace-dev.inc
index 94945ec..74bf493 100644
--- a/libcutils/trace-dev.inc
+++ b/libcutils/trace-dev.inc
@@ -52,7 +52,7 @@ constexpr uint32_t kSeqNoNotInit = static_cast<uint32_t>(-1);
 atomic_bool              atrace_is_ready      = ATOMIC_VAR_INIT(false);
 int                      atrace_marker_fd     = -1;
 uint64_t                 atrace_enabled_tags  = ATRACE_TAG_NOT_READY;
-static atomic_bool       atrace_is_enabled    = ATOMIC_VAR_INIT(true);
+static atomic_bool       atrace_is_enabled    = ATOMIC_VAR_INIT(false);
 static pthread_mutex_t   atrace_tags_mutex    = PTHREAD_MUTEX_INITIALIZER;

 /**

diff --git a/rootdir/init.rc b/rootdir/init.rc
index 2b53d88..e0b8916 100644
--- a/rootdir/init.rc
+++ b/rootdir/init.rc
@@ -676,11 +676,6 @@ on post-fs-data
     # Make sure we have the device encryption key.
     installkey /data

-    # Start bootcharting as soon as possible after the data partition is
-    # mounted to collect more data.
-    mkdir /data/bootchart 0755 shell shell encryption=Require
-    bootchart start
-
     # Avoid predictable entropy pool. Carry over entropy from previous boot.
     copy /data/system/entropy.dat /dev/urandom

@@ -689,13 +684,6 @@ on post-fs-data
     mkdir /data/vendor_de 0771 root root encryption=None
     mkdir /data/vendor/hardware 0771 root root

-    # Start tombstoned early to be able to store tombstones.
-    mkdir /data/anr 0775 system system encryption=Require
-    mkdir /data/tombstones 0771 system system encryption=Require
-    mkdir /data/vendor/tombstones 0771 root root
-    mkdir /data/vendor/tombstones/wifi 0771 wifi wifi
-    start tombstoned
-
     # Make sure that apexd is started in the default namespace
     enter_default_mount_ns

@@ -772,7 +760,6 @@ on post-fs-data
     mkdir /data/misc/bluetooth 0770 bluetooth bluetooth
     mkdir /data/misc/bluetooth/logs 0770 bluetooth bluetooth
     mkdir /data/misc/nfc 0770 nfc nfc
-    mkdir /data/misc/nfc/logs 0770 nfc nfc
     mkdir /data/misc/credstore 0700 credstore credstore
     mkdir /data/misc/gatekeeper 0700 system system
     mkdir /data/misc/keychain 0771 system system
@@ -801,14 +788,6 @@ on post-fs-data
     mkdir /data/misc/audioserver 0700 audioserver audioserver
     mkdir /data/misc/cameraserver 0700 cameraserver cameraserver
     mkdir /data/misc/vold 0700 root root
-    mkdir /data/misc/boottrace 0771 system shell
-    mkdir /data/misc/update_engine 0700 root root
-    mkdir /data/misc/update_engine_log 02750 root log
-    mkdir /data/misc/trace 0700 root root
-    # create location to store surface and window trace files
-    mkdir /data/misc/wmtrace 0700 system system
-    # create location to store accessibility trace files
-    mkdir /data/misc/a11ytrace 0700 system system
     # profile file layout
     mkdir /data/misc/profiles 0771 system system
     mkdir /data/misc/profiles/cur 0771 system system
@@ -819,7 +798,6 @@ on post-fs-data
     mkdir /data/misc/apexdata 0711 root root
     mkdir /data/misc/apexrollback 0700 root root
     mkdir /data/misc/appcompat/ 0700 system system
-    mkdir /data/misc/snapshotctl_log 0755 root root
     # create location to store pre-reboot information
     mkdir /data/misc/prereboot 0700 system system
     # directory used for on-device refresh metrics file.
@@ -840,7 +818,6 @@ on post-fs-data
     # For security reasons, /data/local/tmp should always be empty.
     # Do not place files or directories in /data/local/tmp
     mkdir /data/local/tmp 0771 shell shell
-    mkdir /data/local/traces 0777 shell shell
     mkdir /data/data 0771 system system encryption=None
     mkdir /data/app-private 0771 system system encryption=Require
     mkdir /data/app-ephemeral 0771 system system encryption=Require
@@ -853,25 +830,8 @@ on post-fs-data
     mkdir /data/fonts/files 0771 system system
     mkdir /data/fonts/config 0770 system system

-    # Create directories to push tests to for each linker namespace.
-    # Create the subdirectories in case the first test is run as root
-    # so it doesn't end up owned by root.
-    # Set directories to be executable by any process so that debuggerd,
-    # aka crash_dump, can read any executables/shared libraries.
-    mkdir /data/local/tests 0701 shell shell
-    mkdir /data/local/tests/product 0701 shell shell
-    mkdir /data/local/tests/system 0701 shell shell
-    mkdir /data/local/tests/unrestricted 0701 shell shell
-    mkdir /data/local/tests/vendor 0701 shell shell
-
     # create dalvik-cache, so as to enforce our permissions
     mkdir /data/dalvik-cache 0771 root root encryption=Require
-    # create the A/B OTA directory, so as to enforce our permissions
-    mkdir /data/ota 0771 root root encryption=Require
-
-    # create the OTA package directory. It will be accessed by GmsCore (cache
-    # group), update_engine and update_verifier.
-    mkdir /data/ota_package 0770 system cache encryption=Require

     # create resource-cache and double-check the perms
     mkdir /data/resource-cache 0771 system system encryption=Require
@@ -902,8 +862,6 @@ on post-fs-data
     mkdir /data/system/environ 0700 system system
     # b/183861600 attempt to fix selinux label before running derive_classpath service
     restorecon /data/system/environ
-    mkdir /data/system/dropbox 0700 system system
-    mkdir /data/system/heapdump 0700 system system
     mkdir /data/system/users 0775 system system

     mkdir /data/system_de 0770 system system encryption=None
@@ -949,10 +907,7 @@ on post-fs-data
     mkdir /data_mirror/ref_profiles 0700 root root
     mount none /data/misc/profiles/ref /data_mirror/ref_profiles bind rec

-    mkdir /data/cache 0770 system cache encryption=Require
-    mkdir /data/cache/recovery 0770 system cache
-    mkdir /data/cache/backup_stage 0700 system system
-    mkdir /data/cache/backup 0700 system system
+    symlink /cache /data/cache

     # Delete these if need be, per b/139193659
     mkdir /data/rollback 0700 system system encryption=DeleteIfNecessary
@@ -1192,7 +1147,6 @@ on charger
     class_start charger

 on property:sys.boot_completed=1
-    bootchart stop
     # Setup per_boot directory so other .rc could start to use it on boot_completed
     exec - system system -- /bin/rm -rf /data/per_boot
     mkdir /data/per_boot 0700 system system encryption=Require key=per_boot_ref
@@ -1260,29 +1214,6 @@ service ueventd /system/bin/ueventd
     seclabel u:r:ueventd:s0
     shutdown critical

-service console /system/bin/sh
-    class core
-    console
-    disabled
-    user shell
-    group shell log readproc
-    seclabel u:r:shell:s0
-    setenv HOSTNAME console
-
-on property:ro.debuggable=1
-    # Give writes to the same group for the trace folder on debug builds,
-    # it's further protected by selinux policy.
-    # The folder is used to store method traces.
-    chmod 0773 /data/misc/trace
-    # Give writes and reads to anyone for the window trace folder on debug builds,
-    # it's further protected by selinux policy.
-    chmod 0777 /data/misc/wmtrace
-    # Give reads to anyone for the accessibility trace folder on debug builds.
-    chmod 0775 /data/misc/a11ytrace
-
-on init && property:ro.debuggable=1
-    start console
-
 on userspace-reboot-requested
   # TODO(b/135984674): reset all necessary properties here.
   setprop sys.boot_completed ""

project system/sepolicy/
diff --git a/Android.bp b/Android.bp
index 8e2a966..e99fb54 100644
--- a/Android.bp
+++ b/Android.bp
@@ -868,7 +868,6 @@ precompiled_se_policy_binary {
     },
     required: [
         "sepolicy_neverallows",
-        "sepolicy_neverallows_vendor",
     ],
     dist: {
         targets: ["base-sepolicy-files-for-mapping"],

diff --git a/Android.mk b/Android.mk
index 50c265d..f743552 100644
--- a/Android.mk
+++ b/Android.mk
@@ -540,7 +540,6 @@ endif # ifdef HAS_PRODUCT_SEPOLICY

 built_sepolicy := $(call intermediates-dir-for,ETC,precompiled_sepolicy)/precompiled_sepolicy
 built_sepolicy_neverallows := $(call intermediates-dir-for,ETC,sepolicy_neverallows)/sepolicy_neverallows
-built_sepolicy_neverallows += $(call intermediates-dir-for,ETC,sepolicy_neverallows_vendor)/sepolicy_neverallows_vendor

 #################################
 # sepolicy is also built with Android.bp.

diff --git a/build/soong/policy.go b/build/soong/policy.go
index b1840da..fee9a0e 100644
--- a/build/soong/policy.go
+++ b/build/soong/policy.go
@@ -379,9 +379,7 @@ func (c *policyCil) compileConfToCil(ctx android.ModuleContext, conf android.Pat
 			FlagWithArg("-o ", os.DevNull).
 			FlagWithArg("-f ", os.DevNull)

-		if proptools.BoolDefault(c.properties.Ignore_neverallow, ctx.Config().SelinuxIgnoreNeverallows()) {
-			secilcCmd.Flag("-N")
-		}
+		secilcCmd.Flag("-N")
 	}

 	rule.Build("cil", "Building cil for "+ctx.ModuleName())
@@ -494,9 +492,7 @@ func (c *policyBinary) GenerateAndroidBuildActions(ctx android.ModuleContext) {
 		FlagWithOutput("-o ", bin).
 		FlagWithArg("-f ", os.DevNull)

-	if proptools.BoolDefault(c.properties.Ignore_neverallow, ctx.Config().SelinuxIgnoreNeverallows()) {
-		secilcCmd.Flag("-N")
-	}
+	secilcCmd.Flag("-N")
 	rule.Temporary(bin)

 	// permissive check is performed only in user build (not debuggable).
@@ -510,7 +506,7 @@ func (c *policyBinary) GenerateAndroidBuildActions(ctx android.ModuleContext) {
 		rule.Temporary(permissiveDomains)

 		msg := `==========\n` +
-			`ERROR: permissive domains not allowed in user builds\n` +
+			`WARNING: permissive domains not allowed in user builds\n` +
 			`List of invalid domains:`

 		rule.Command().Text("if test").
@@ -520,7 +516,7 @@ func (c *policyBinary) GenerateAndroidBuildActions(ctx android.ModuleContext) {
 			Text(`"` + msg + `"`).
 			Text("&& cat ").
 			Input(permissiveDomains).
-			Text("; exit 1; fi")
+			Text("; fi")
 	}

 	out := android.PathForModuleOut(ctx, c.stem())

project vendor/lineage/
diff --git a/bootanimation/Android.mk b/bootanimation/Android.mk
index f58a850..853e770 100644
--- a/bootanimation/Android.mk
+++ b/bootanimation/Android.mk
@@ -62,5 +62,3 @@ BOOTANIMATION_SYMLINK := $(TARGET_OUT_PRODUCT)/media/bootanimation-dark.zip
 $(BOOTANIMATION_SYMLINK): $(LOCAL_INSTALLED_MODULE)
 	@mkdir -p $(dir $@)
 	$(hide) ln -sf bootanimation.zip $@
-
-ALL_DEFAULT_INSTALLED_MODULES += $(BOOTANIMATION_SYMLINK)

diff --git a/config/common.mk b/config/common.mk
index 7947854..68faef2 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -102,8 +102,6 @@ endif
 # Bootanimation
 TARGET_SCREEN_WIDTH ?= 1080
 TARGET_SCREEN_HEIGHT ?= 1920
-PRODUCT_PACKAGES += \
-    bootanimation.zip

 # Build Manifest
 PRODUCT_PACKAGES += \
@@ -113,31 +111,12 @@ PRODUCT_PACKAGES += \
 PRODUCT_PACKAGES += \
     LineageParts \
     LineageSettingsProvider \
-    LineageSetupWizard \
-    Updater
-
-PRODUCT_COPY_FILES += \
-    vendor/lineage/prebuilt/common/etc/init/init.lineage-updater.rc:$(TARGET_COPY_OUT_SYSTEM_EXT)/etc/init/init.lineage-updater.rc
+    LineageSetupWizard

 # Config
 PRODUCT_PACKAGES += \
     SimpleDeviceConfig

-# Extra tools in Lineage
-PRODUCT_PACKAGES += \
-    bash \
-    curl \
-    getcap \
-    htop \
-    nano \
-    setcap \
-    vim
-
-PRODUCT_ARTIFACT_PATH_REQUIREMENT_ALLOWED_LIST += \
-    system/bin/curl \
-    system/bin/getcap \
-    system/bin/setcap
-
 # Filesystems tools
 PRODUCT_PACKAGES += \
     fsck.ntfs \
@@ -151,23 +130,6 @@ PRODUCT_ARTIFACT_PATH_REQUIREMENT_ALLOWED_LIST += \
     system/%/libfuse-lite.so \
     system/%/libntfs-3g.so

-# Openssh
-PRODUCT_PACKAGES += \
-    scp \
-    sftp \
-    ssh \
-    sshd \
-    sshd_config \
-    ssh-keygen \
-    start-ssh
-
-PRODUCT_COPY_FILES += \
-    vendor/lineage/prebuilt/common/etc/init/init.openssh.rc:$(TARGET_COPY_OUT_PRODUCT)/etc/init/init.openssh.rc
-
-# rsync
-PRODUCT_PACKAGES += \
-    rsync
-
 # Storage manager
 PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
     ro.storage_manager.enabled=true

diff --git a/config/common_full.mk b/config/common_full.mk
index f7eecd8..48dcf9b 100644
--- a/config/common_full.mk
+++ b/config/common_full.mk
@@ -3,35 +3,13 @@ $(call inherit-product, vendor/lineage/config/common_mobile.mk)

 PRODUCT_SIZE := full

-# Include {Lato,Rubik} fonts
-$(call inherit-product-if-exists, external/google-fonts/lato/fonts.mk)
-$(call inherit-product-if-exists, external/google-fonts/rubik/fonts.mk)
-
 # Apps
 PRODUCT_PACKAGES += \
     Eleven \
     Etar \
-    Profiles \
-    Recorder \
-    Seedvault
+    Recorder

 ifneq ($(PRODUCT_NO_CAMERA),true)
 PRODUCT_PACKAGES += \
     Aperture
 endif
-
-ifneq ($(TARGET_EXCLUDES_AUDIOFX),true)
-PRODUCT_PACKAGES += \
-    AudioFX
-endif
-
-# Extra cmdline tools
-PRODUCT_PACKAGES += \
-    unrar \
-    zstd
-
-# Fonts
-PRODUCT_PACKAGES += \
-    fonts_customization.xml \
-    FontLatoOverlay \
-    FontRubikOverlay

diff --git a/config/common_mobile.mk b/config/common_mobile.mk
index a9a66ee..da2c8a8 100644
--- a/config/common_mobile.mk
+++ b/config/common_mobile.mk
@@ -47,38 +47,9 @@ endif

 # Customizations
 PRODUCT_PACKAGES += \
-    IconPackCircularAndroidOverlay \
-    IconPackCircularLauncherOverlay \
-    IconPackCircularSettingsOverlay \
-    IconPackCircularSystemUIOverlay \
-    IconPackFilledAndroidOverlay \
-    IconPackFilledLauncherOverlay \
-    IconPackFilledSettingsOverlay \
-    IconPackFilledSystemUIOverlay \
-    IconPackKaiAndroidOverlay \
-    IconPackKaiLauncherOverlay \
-    IconPackKaiSettingsOverlay \
-    IconPackKaiSystemUIOverlay \
-    IconPackRoundedAndroidOverlay \
-    IconPackRoundedLauncherOverlay \
-    IconPackRoundedSettingsOverlay \
-    IconPackRoundedSystemUIOverlay \
-    IconPackSamAndroidOverlay \
-    IconPackSamLauncherOverlay \
-    IconPackSamSettingsOverlay \
-    IconPackSamSystemUIOverlay \
-    IconPackVictorAndroidOverlay \
-    IconPackVictorLauncherOverlay \
-    IconPackVictorSettingsOverlay \
-    IconPackVictorSystemUIOverlay \
-    IconShapePebbleOverlay \
-    IconShapeRoundedRectOverlay \
-    IconShapeSquareOverlay \
-    IconShapeSquircleOverlay \
-    IconShapeTaperedRectOverlay \
-    IconShapeTeardropOverlay \
-    IconShapeVesselOverlay \
     LineageNavigationBarNoHint \
+    NavigationBarMode3ButtonOverlay \
+    NavigationBarModeGesturalOverlay \
     NavigationBarMode2ButtonOverlay

 # Media
@@ -91,6 +62,4 @@ PRODUCT_PACKAGES += \

 # Themes
 PRODUCT_PACKAGES += \
-    LineageBlackTheme \
-    ThemePicker \
-    ThemesStub
+    ThemePicker

diff --git a/config/telephony.mk b/config/telephony.mk
index be414ed..a75ba9a 100644
--- a/config/telephony.mk
+++ b/config/telephony.mk
@@ -13,8 +13,7 @@ PRODUCT_PACKAGES += \

 # Telephony packages
 PRODUCT_PACKAGES += \
-    messaging \
-    Stk
+    messaging

 # Default ringtone
 PRODUCT_PRODUCT_PROPERTIES += \

diff --git a/prebuilt/common/etc/init/init.lineage-system_ext.rc b/prebuilt/common/etc/init/init.lineage-system_ext.rc
index fdee7ea..d75d559 100644
--- a/prebuilt/common/etc/init/init.lineage-system_ext.rc
+++ b/prebuilt/common/etc/init/init.lineage-system_ext.rc
@@ -1,9 +1,3 @@
 # LineageOS core functionality
 on init
     export TERMINFO /system_ext/etc/terminfo
-
-# bugreport is triggered by holding down volume down, volume up and power
-service bugreport /system/bin/dumpstate -d -p -z
-    oneshot
-    disabled
-    keycodes 114 115 116
