
project build/make/
diff --git a/core/Makefile b/core/Makefile
index 20db344..8e2fa52 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -7183,7 +7183,7 @@ INTERNAL_OTA_METADATA := $(PRODUCT_OUT)/ota_metadata
 $(call declare-0p-target,$(INTERNAL_OTA_METADATA))
 
 ifeq ($(TARGET_BUILD_VARIANT),user)
-    $(INTERNAL_OTA_PACKAGE_TARGET): backuptool := false
+    $(INTERNAL_OTA_PACKAGE_TARGET): backuptool := true
 else
 ifneq ($(LINEAGE_BUILD),)
     $(INTERNAL_OTA_PACKAGE_TARGET): backuptool := true
diff --git a/core/java.mk b/core/java.mk
index 5fbc916..a68057d 100644
--- a/core/java.mk
+++ b/core/java.mk
@@ -210,7 +210,7 @@ $(full_classes_compiled_jar): PRIVATE_WARNINGS_ENABLE := $(LOCAL_WARNINGS_ENABLE
 # available via JDWP.
 ifneq (,$(PRODUCT_MINIMIZE_JAVA_DEBUG_INFO))
 ifneq (,$(filter userdebug user,$(TARGET_BUILD_VARIANT)))
-LOCAL_JAVACFLAGS+= -g:source,lines
+LOCAL_JAVACFLAGS+= -g:none
 endif
 endif
 
diff --git a/core/sysprop.mk b/core/sysprop.mk
index e7a140a..5126189 100644
--- a/core/sysprop.mk
+++ b/core/sysprop.mk
@@ -173,7 +173,7 @@ endif
 # Both of these tags will be removed and replaced with "release-keys"
 # when the target-files is signed in a post-build step.
 ifeq ($(DEFAULT_SYSTEM_DEV_CERTIFICATE),build/make/target/product/security/testkey)
-BUILD_KEYS := test-keys
+BUILD_KEYS := release-keys
 else
 BUILD_KEYS := dev-keys
 endif
diff --git a/target/product/base_system.mk b/target/product/base_system.mk
index 339a20b..79608c3 100644
--- a/target/product/base_system.mk
+++ b/target/product/base_system.mk
@@ -31,9 +31,7 @@ PRODUCT_PACKAGES += \
     appops \
     app_process \
     appwidget \
-    atrace \
     audioserver \
-    BackupRestoreConfirmation \
     bcc \
     blank_screen \
     blkid \
@@ -43,8 +41,6 @@ PRODUCT_PACKAGES += \
     boringssl_self_test \
     bpfloader \
     bu \
-    bugreport \
-    bugreportz \
     build_flag_system \
     cgroups.json \
     charger \
@@ -79,9 +75,6 @@ PRODUCT_PACKAGES += \
     com.android.wifi \
     ContactsProvider \
     content \
-    CtsShimPrebuilt \
-    CtsShimPrivPrebuilt \
-    debuggerd\
     device_config \
     dmctl \
     dnsmasq \
@@ -110,7 +103,6 @@ PRODUCT_PACKAGES += \
     group_system \
     gsid \
     gsi_tool \
-    heapprofd \
     heapprofd_client \
     gatekeeperd \
     gpuservice \
@@ -286,9 +278,6 @@ PRODUCT_PACKAGES += \
     tc \
     telecom \
     telephony-common \
-    tombstoned \
-    traced \
-    traced_probes \
     tune2fs \
     uiautomator \
     uinput \
@@ -328,13 +317,6 @@ ifneq ($(PRODUCT_IS_ATV),true)
 
 endif
 
-# Product does not support Dynamic System Update
-ifneq ($(PRODUCT_NO_DYNAMIC_SYSTEM_UPDATE),true)
-    PRODUCT_PACKAGES += \
-        DynamicSystemInstallationService \
-
-endif
-
 # Check if the build supports NFC apex or not
 ifeq ($(RELEASE_PACKAGE_NFC_STACK),NfcNci)
     PRODUCT_PACKAGES += \
@@ -400,11 +382,6 @@ ifeq ($(EMMA_INSTRUMENT),true)
   endif # EMMA_INSTRUMENT_STATIC
 endif # EMMA_INSTRUMENT
 
-ifeq (,$(DISABLE_WALLPAPER_BACKUP))
-  PRODUCT_PACKAGES += \
-    WallpaperBackup
-endif
-
 PRODUCT_PACKAGES += \
     libEGL_angle \
     libGLESv1_CM_angle \
@@ -472,9 +449,6 @@ PRODUCT_PACKAGES += etc_hosts
 PRODUCT_PACKAGES += init.zygote32.rc
 PRODUCT_VENDOR_PROPERTIES += ro.zygote?=zygote32
 
-PRODUCT_SYSTEM_PROPERTIES += debug.atrace.tags.enableflags=0
-PRODUCT_SYSTEM_PROPERTIES += persist.traced.enable=1
-
 # Include kernel configs.
 PRODUCT_PACKAGES += \
     approved-ogki-builds.xml \
@@ -524,11 +498,6 @@ endif
 PRODUCT_SYSTEM_SERVER_APPS += \
     SettingsProvider \
 
-ifeq (,$(DISABLE_WALLPAPER_BACKUP))
-  PRODUCT_SYSTEM_SERVER_APPS += \
-    WallpaperBackup
-endif
-
 PRODUCT_PACKAGES_DEBUG_JAVA_COVERAGE := \
     libdumpcoverage
 
diff --git a/target/product/full_base.mk b/target/product/full_base.mk
index 7aa7903..5f5aad1 100644
--- a/target/product/full_base.mk
+++ b/target/product/full_base.mk
@@ -25,7 +25,6 @@ PRODUCT_PACKAGES := \
 
 PRODUCT_PACKAGES += \
     LiveWallpapersPicker \
-    PhotoTable \
     preinstalled-packages-platform-full-base.xml
 
 # Net:
diff --git a/target/product/handheld_product.mk b/target/product/handheld_product.mk
index b2ce4c5..c468988 100644
--- a/target/product/handheld_product.mk
+++ b/target/product/handheld_product.mk
@@ -31,7 +31,6 @@ PRODUCT_PACKAGES += \
     preinstalled-packages-platform-handheld-product.xml \
     QuickSearchBox \
     SettingsIntelligence \
-    frameworks-base-overlays
 
 ifeq ($(LINEAGE_BUILD),)
 PRODUCT_PACKAGES += \
diff --git a/target/product/handheld_system.mk b/target/product/handheld_system.mk
index d61479d..be45725 100644
--- a/target/product/handheld_system.mk
+++ b/target/product/handheld_system.mk
@@ -34,10 +34,8 @@ $(call inherit-product-if-exists, frameworks/webview/chromium/chromium.mk)
 
 PRODUCT_PACKAGES += \
     android.software.window_magnification.prebuilt.xml \
-    BasicDreams \
     BlockedNumberProvider \
     BluetoothMidiService \
-    BookmarkProvider \
     BuiltInPrintService \
     CalendarProvider \
     cameraserver \
@@ -48,7 +46,6 @@ PRODUCT_PACKAGES += \
     DeviceAsWebcam \
     DocumentsUI \
     DownloadProviderUi \
-    EasterEgg \
     ExternalStorageProvider \
     FusedLocation \
     InputDevices \
@@ -57,7 +54,6 @@ PRODUCT_PACKAGES += \
     ManagedProvisioning \
     MmsService \
     MtpService \
-    MusicFX \
     PacProcessor \
     preinstalled-packages-platform-handheld-system.xml \
     PrintRecommendationService \
@@ -65,8 +61,6 @@ PRODUCT_PACKAGES += \
     ProxyHandler \
     screenrecord \
     SecureElement \
-    SharedStorageBackup \
-    SimAppDialog \
     Telecom \
     TeleService \
     UserDictionaryProvider \
diff --git a/target/product/handheld_system_ext.mk b/target/product/handheld_system_ext.mk
index 187b627..5d5f758 100644
--- a/target/product/handheld_system_ext.mk
+++ b/target/product/handheld_system_ext.mk
@@ -28,4 +28,3 @@ PRODUCT_PACKAGES += \
     Settings \
     StorageManager \
     SystemUI \
-    WallpaperCropper \
diff --git a/target/product/media_system.mk b/target/product/media_system.mk
index 0733d06..9448cbf 100644
--- a/target/product/media_system.mk
+++ b/target/product/media_system.mk
@@ -64,10 +64,6 @@ PRODUCT_VENDOR_PROPERTIES += \
     tombstoned.max_tombstone_count?=50
 endif
 
-PRODUCT_VENDOR_PROPERTIES += \
-    ro.logd.size.stats=64K \
-    log.tag.stats_log=I
-
 # Enable CFI for security-sensitive components
 $(call inherit-product, $(SRC_TARGET_DIR)/product/cfi-common.mk)
 $(call inherit-product-if-exists, vendor/google/products/cfi-vendor.mk)
diff --git a/target/product/telephony_system.mk b/target/product/telephony_system.mk
index ef48719..4b76ad5 100644
--- a/target/product/telephony_system.mk
+++ b/target/product/telephony_system.mk
@@ -20,8 +20,5 @@
 PRODUCT_PACKAGES := \
     ONS \
     CarrierDefaultApp \
-    CallLogBackup \
-    com.android.cellbroadcast \
-    CellBroadcastLegacyApp \
 
 PRODUCT_COPY_FILES := \
diff --git a/tools/releasetools/common.py b/tools/releasetools/common.py
index 2afa3b9..ce07d2f 100644
--- a/tools/releasetools/common.py
+++ b/tools/releasetools/common.py
@@ -3863,10 +3863,7 @@ def MakeRecoveryPatch(input_dir, output_sink, recovery_img, boot_img,
     logger.warning('Recovery patch generation is disable when prebuilt vendor image is used.')
     return None
 
-  if full_recovery_image:
-    output_sink(recovery_img_path, recovery_img.data)
-
-  else:
+  if not full_recovery_image:
     d = Difference(recovery_img, boot_img)
     _, _, patch = d.ComputePatch()
     output_sink("recovery-from-boot.p", patch)
@@ -3925,7 +3922,6 @@ fi
 
   # The install script location moved from /system/etc to /system/bin in the L
   # release. In the R release it is in VENDOR/bin or SYSTEM/vendor/bin.
-  output_sink("bin/install-recovery.sh", sh.encode())
 
 
 class DynamicPartitionUpdate(object):
diff --git a/tools/releasetools/non_ab_ota.py b/tools/releasetools/non_ab_ota.py
index bc98187..cce8fb8 100644
--- a/tools/releasetools/non_ab_ota.py
+++ b/tools/releasetools/non_ab_ota.py
@@ -149,8 +149,6 @@ def WriteFullOTAPackage(input_zip, output_file):
       metadata=metadata,
       info_dict=OPTIONS.info_dict)
 
-  assert HasRecoveryPatch(input_zip, info_dict=OPTIONS.info_dict)
-
   # Assertions (e.g. downgrade check, device properties check).
   #ts = target_info.GetBuildProp("ro.build.date.utc")
   #ts_text = target_info.GetBuildProp("ro.build.date")
@@ -184,10 +182,6 @@ def WriteFullOTAPackage(input_zip, output_file):
   #    complete script normally
   #    (allow recovery to mark itself finished and reboot)
 
-  recovery_img = common.GetBootableImage("recovery.img", "recovery.img",
-                                         OPTIONS.input_tmp, "RECOVERY")
-  common.CheckSize(recovery_img.data, "recovery.img", target_info)
-  common.ZipWriteStr(output_zip, "recovery.img", recovery_img.data)
   if OPTIONS.two_step:
     if not target_info.get("multistage_support"):
       assert False, "two-step packages not supported by this build"

project build/soong/
diff --git a/scripts/gen_build_prop.py b/scripts/gen_build_prop.py
index ac2c6ff..a4dd8cc 100644
--- a/scripts/gen_build_prop.py
+++ b/scripts/gen_build_prop.py
@@ -42,7 +42,7 @@ def get_build_flavor(product_config):
 def get_build_keys(product_config):
   default_cert = product_config.get("DefaultAppCertificate", "")
   if default_cert == "" or default_cert == os.path.join(TEST_KEY_DIR, "testKey"):
-    return "test-keys"
+    return "release-keys"
   return "dev-keys"
 
 def override_config(config):
diff --git a/ui/build/finder.go b/ui/build/finder.go
index 573df21..50aba12 100644
--- a/ui/build/finder.go
+++ b/ui/build/finder.go
@@ -65,7 +65,7 @@ func NewSourceFinder(ctx Context, config Config) (f *finder.Finder) {
 		WorkingDirectory: dir,
 		RootDirs:         androidBpSearchDirs(config),
 		FollowSymlinks:   config.environ.IsEnvTrue("ALLOW_BP_UNDER_SYMLINKS"),
-		ExcludeDirs:      []string{".git", ".repo"},
+		ExcludeDirs:      []string{".git", ".repo", "KernelSU"},
 		PruneFiles:       pruneFiles,
 		IncludeFiles: []string{
 			// Kati build definitions.

project frameworks/base/
diff --git a/services/core/java/com/android/server/pm/ComputerEngine.java b/services/core/java/com/android/server/pm/ComputerEngine.java
index 4cf099aed..56f87f1fc 100644
--- a/services/core/java/com/android/server/pm/ComputerEngine.java
+++ b/services/core/java/com/android/server/pm/ComputerEngine.java
@@ -1491,10 +1491,6 @@ public class ComputerEngine implements Computer {
     private static native boolean isDebuggable();
 
     public static boolean isMicrogSigned(AndroidPackage p) {
-        if (!isDebuggable()) {
-            return false;
-        }
-
         // Allowlist the following apps:
         // * com.android.vending - microG Companion
         // * com.google.android.gms - microG Services

project packages/modules/adb/
diff --git a/daemon/main.cpp b/daemon/main.cpp
index 6f0fcbc..4b6c074 100644
--- a/daemon/main.cpp
+++ b/daemon/main.cpp
@@ -72,7 +72,7 @@ static bool should_drop_privileges() {
     // ro.secure:
     //   Drop privileges by default. Set to 1 on userdebug and user builds.
     bool ro_secure = android::base::GetBoolProperty("ro.secure", true);
-    bool ro_debuggable = __android_log_is_debuggable();
+    bool ro_debuggable = true;
 
     // Drop privileges if ro.secure is set...
     bool drop = ro_secure;
@@ -118,7 +118,7 @@ static void drop_privileges(int server_port) {
     // Don't listen on a port (default 5037) if running in secure mode.
     // Don't run as root if running in secure mode.
     if (should_drop_privileges()) {
-        const bool should_drop_caps = !__android_log_is_debuggable();
+        const bool should_drop_caps = false;
 
         if (should_drop_caps) {
             // CAP_SETUI and CAP_SETGID are required for change_uid and change_gid calls below.
diff --git a/root/adbroot_service.cpp b/root/adbroot_service.cpp
index ff30f4b..243f18e 100644
--- a/root/adbroot_service.cpp
+++ b/root/adbroot_service.cpp
@@ -106,7 +106,7 @@ ndk::ScopedAStatus ADBRootService::isSupported(bool* _aidl_return) {
     }
 
     AutoMutex _l(lock_);
-    *_aidl_return = __android_log_is_debuggable();
+    *_aidl_return = true;
     return ndk::ScopedAStatus::ok();
 }
 
diff --git a/sockets.cpp b/sockets.cpp
index f766f7e..1dd0dca 100644
--- a/sockets.cpp
+++ b/sockets.cpp
@@ -473,7 +473,7 @@ asocket* create_local_service_socket(std::string_view name, atransport* transpor
     LOG(VERBOSE) << "LS(" << s->id << "): bound to '" << name << "' via " << fd_value;
 
 #if !ADB_HOST
-    if ((name.starts_with("root:") && getuid() != 0 && __android_log_is_debuggable()) ||
+    if ((name.starts_with("root:") && getuid() != 0) ||
         (name.starts_with("unroot:") && getuid() == 0) || name.starts_with("usb:") ||
         name.starts_with("tcpip:")) {
         D("LS(%d): enabling exit_on_close", s->id);

project system/core/
diff --git a/init/Android.bp b/init/Android.bp
index 072568d..605cbab 100644
--- a/init/Android.bp
+++ b/init/Android.bp
@@ -114,7 +114,7 @@ libinit_cc_defaults {
     cflags: [
         "-DALLOW_FIRST_STAGE_CONSOLE=0",
         "-DALLOW_LOCAL_PROP_OVERRIDE=0",
-        "-DALLOW_PERMISSIVE_SELINUX=0",
+        "-DALLOW_PERMISSIVE_SELINUX=1",
         "-DANDROID_BASE_UNIQUE_FD_DISABLE_IMPLICIT_CONVERSION",
         "-DDUMP_ON_UMOUNT_FAILURE=0",
         "-DINIT_FULL_SOURCES",
@@ -412,7 +412,7 @@ init_first_stage_cc_defaults {
         "-Werror",
         "-DALLOW_FIRST_STAGE_CONSOLE=0",
         "-DALLOW_LOCAL_PROP_OVERRIDE=0",
-        "-DALLOW_PERMISSIVE_SELINUX=0",
+        "-DALLOW_PERMISSIVE_SELINUX=1",
         "-DREBOOT_BOOTLOADER_ON_PANIC=0",
         "-DWORLD_WRITABLE_KMSG=0",
         "-DDUMP_ON_UMOUNT_FAILURE=0",
diff --git a/rootdir/init.rc b/rootdir/init.rc
index 1acd637..42f0ac3 100644
--- a/rootdir/init.rc
+++ b/rootdir/init.rc
@@ -709,24 +709,12 @@ on post-fs-data
     # Make sure we have the device encryption key.
     installkey /data
 
-    # Start bootcharting as soon as possible after the data partition is
-    # mounted to collect more data.
-    mkdir /data/bootchart 0755 shell shell encryption=Require
-    bootchart start
-
     # Avoid predictable entropy pool. Carry over entropy from previous boot.
     copy /data/system/entropy.dat /dev/urandom
 
     mkdir /data/vendor 0771 root root encryption=Require
     mkdir /data/vendor/hardware 0771 root root
 
-    # Start tombstoned early to be able to store tombstones.
-    mkdir /data/anr 0775 system system encryption=Require
-    mkdir /data/tombstones 0775 system system encryption=Require
-    mkdir /data/vendor/tombstones 0771 root root
-    mkdir /data/vendor/tombstones/wifi 0771 wifi wifi
-    start tombstoned
-
     # Make sure that apexd is started in the default namespace
     enter_default_mount_ns
 
@@ -807,9 +795,7 @@ on post-fs-data
     chmod 0660 /data/misc/bluedroid/bt_config.conf
     chown bluetooth bluetooth /data/misc/bluedroid/bt_config.conf
     mkdir /data/misc/bluetooth 0770 bluetooth bluetooth
-    mkdir /data/misc/bluetooth/logs 0770 bluetooth bluetooth
     mkdir /data/misc/nfc 0770 nfc nfc
-    mkdir /data/misc/nfc/logs 0770 nfc nfc
     mkdir /data/misc/credstore 0700 credstore credstore
     mkdir /data/misc/gatekeeper 0700 system system
     mkdir /data/misc/keychain 0771 system system
@@ -838,14 +824,6 @@ on post-fs-data
     mkdir /data/misc/audioserver 0700 audioserver audioserver
     mkdir /data/misc/cameraserver 0700 cameraserver cameraserver
     mkdir /data/misc/vold 0700 root root
-    mkdir /data/misc/boottrace 0771 system shell
-    mkdir /data/misc/update_engine 0700 root root
-    mkdir /data/misc/update_engine_log 02750 root log
-    mkdir /data/misc/trace 0700 root root
-    # create location to store surface and window trace files
-    mkdir /data/misc/wmtrace 0700 system system
-    # create location to store accessibility trace files
-    mkdir /data/misc/a11ytrace 0700 system system
     # profile file layout
     mkdir /data/misc/profiles 0771 system system
     mkdir /data/misc/profiles/cur 0771 system system
@@ -857,7 +835,6 @@ on post-fs-data
     mkdir /data/misc/apexrollback 0700 root root
     mkdir /data/misc/appcompat/ 0700 system system
     mkdir /data/misc/uprobestats-configs/ 0777 uprobestats uprobestats
-    mkdir /data/misc/snapshotctl_log 0755 root root
     # create location to store pre-reboot information
     mkdir /data/misc/prereboot 0700 system system
     # directory used for on-device refresh metrics file.
@@ -882,7 +859,6 @@ on post-fs-data
     # For security reasons, /data/local/tmp should always be empty.
     # Do not place files or directories in /data/local/tmp
     mkdir /data/local/tmp 0771 shell shell
-    mkdir /data/local/traces 0777 shell shell
     mkdir /data/app-private 0771 system system encryption=Require
     mkdir /data/app-ephemeral 0771 system system encryption=Require
     mkdir /data/app-asec 0700 root root encryption=Require
@@ -897,25 +873,8 @@ on post-fs-data
     mkdir /data/fonts/files 0771 system system
     mkdir /data/fonts/config 0770 system system
 
-    # Create directories to push tests to for each linker namespace.
-    # Create the subdirectories in case the first test is run as root
-    # so it doesn't end up owned by root.
-    # Set directories to be executable by any process so that debuggerd,
-    # aka crash_dump, can read any executables/shared libraries.
-    mkdir /data/local/tests 0701 shell shell
-    mkdir /data/local/tests/product 0701 shell shell
-    mkdir /data/local/tests/system 0701 shell shell
-    mkdir /data/local/tests/unrestricted 0701 shell shell
-    mkdir /data/local/tests/vendor 0701 shell shell
-
     # create dalvik-cache, so as to enforce our permissions
     mkdir /data/dalvik-cache 0771 root root encryption=Require
-    # create the A/B OTA directory, so as to enforce our permissions
-    mkdir /data/ota 0771 root root encryption=Require
-
-    # create the OTA package directory. It will be accessed by GmsCore (cache
-    # group), update_engine and update_verifier.
-    mkdir /data/ota_package 0770 system cache encryption=Require
 
     # create resource-cache and double-check the perms
     mkdir /data/resource-cache 0771 system system encryption=Require
@@ -949,13 +908,7 @@ on post-fs-data
     mkdir /data/system/environ 0700 system system
     # b/183861600 attempt to fix selinux label before running derive_classpath service
     restorecon /data/system/environ
-    mkdir /data/system/dropbox 0700 system system
-    mkdir /data/system/heapdump 0700 system system
     mkdir /data/system/users 0775 system system
-    # Mkdir and set SELinux security contexts for shutdown-checkpoints.
-    # TODO(b/270286197): remove these after couple releases.
-    mkdir /data/system/shutdown-checkpoints 0700 system system
-    restorecon_recursive /data/system/shutdown-checkpoints
 
     # Create the parent directories of the user CE and DE storage directories.
     # These parent directories must use encryption=None, since each of their
@@ -1026,10 +979,7 @@ on post-fs-data
     mkdir /data_mirror/ref_profiles 0700 root root
     mount none /data/misc/profiles/ref /data_mirror/ref_profiles bind rec
 
-    mkdir /data/cache 0770 system cache encryption=Require
-    mkdir /data/cache/recovery 0770 system cache
-    mkdir /data/cache/backup_stage 0700 system system
-    mkdir /data/cache/backup 0700 system system
+    symlink /cache /data/cache
 
     # Delete these if need be, per b/139193659
     mkdir /data/rollback 0700 system system encryption=DeleteIfNecessary
@@ -1265,7 +1215,6 @@ on charger
     class_start charger
 
 on property:sys.boot_completed=1
-    bootchart stop
     # Setup per_boot directory so other .rc could start to use it on boot_completed
     exec - system system -- /bin/rm -rf /data/per_boot
     mkdir /data/per_boot 0700 system system encryption=Require key=per_boot_ref
@@ -1344,17 +1293,6 @@ service console /system/bin/sh
     setenv HOSTNAME console
     shutdown critical
 
-on property:ro.debuggable=1
-    # Give writes to the same group for the trace folder on debug builds,
-    # it's further protected by selinux policy.
-    # The folder is used to store method traces.
-    chmod 0773 /data/misc/trace
-    # Give writes and reads to anyone for the window trace folder on debug builds,
-    # it's further protected by selinux policy.
-    chmod 0777 /data/misc/wmtrace
-    # Give reads to anyone for the accessibility trace folder on debug builds.
-    chmod 0775 /data/misc/a11ytrace
-
 on init && property:ro.debuggable=1
     start console
 

project system/sepolicy/
diff --git a/build/soong/policy.go b/build/soong/policy.go
index 8bdf01b..8fd7fa3 100644
--- a/build/soong/policy.go
+++ b/build/soong/policy.go
@@ -414,7 +414,7 @@ func (c *policyCil) compileConfToCil(ctx android.ModuleContext, conf android.Pat
 			FlagWithArg("-o ", os.DevNull).
 			FlagWithArg("-f ", os.DevNull)
 
-		if proptools.BoolDefault(c.properties.Ignore_neverallow, ctx.Config().SelinuxIgnoreNeverallows()) {
+		if proptools.BoolDefault(c.properties.Ignore_neverallow, true) {
 			secilcCmd.Flag("-N")
 		}
 	}
@@ -525,13 +525,13 @@ func (c *policyBinary) GenerateAndroidBuildActions(ctx android.ModuleContext) {
 		FlagWithOutput("-o ", bin).
 		FlagWithArg("-f ", os.DevNull)
 
-	if proptools.BoolDefault(c.properties.Ignore_neverallow, ctx.Config().SelinuxIgnoreNeverallows()) {
+	if proptools.BoolDefault(c.properties.Ignore_neverallow, true) {
 		secilcCmd.Flag("-N")
 	}
 	rule.Temporary(bin)
 
 	// permissive check is performed only in user build (not debuggable).
-	if !ctx.Config().Debuggable() {
+	if false {
 		permissiveDomains := pathForModuleOut(ctx, c.stem()+"_permissive")
 		cmd := rule.Command().BuiltTool("sepolicy-analyze").
 			Input(bin).

project vendor/lineage/
diff --git a/bootanimation/Android.mk b/bootanimation/Android.mk
index 97948f3..8578b28 100644
--- a/bootanimation/Android.mk
+++ b/bootanimation/Android.mk
@@ -59,5 +59,3 @@ BOOTANIMATION_SYMLINK := $(TARGET_OUT_PRODUCT)/media/bootanimation-dark.zip
 $(BOOTANIMATION_SYMLINK): $(LOCAL_INSTALLED_MODULE)
 	@mkdir -p $(dir $@)
 	$(hide) ln -sf bootanimation.zip $@
-
-ALL_DEFAULT_INSTALLED_MODULES += $(BOOTANIMATION_SYMLINK)
diff --git a/config/common.mk b/config/common.mk
index afc8723..aebc7ec 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -115,8 +115,6 @@ endif
 # Bootanimation
 TARGET_SCREEN_WIDTH ?= 1080
 TARGET_SCREEN_HEIGHT ?= 1920
-PRODUCT_PACKAGES += \
-    bootanimation.zip
 
 # Build Manifest
 PRODUCT_PACKAGES += \
@@ -137,10 +135,6 @@ endif
 
 PRODUCT_PACKAGES += \
     LineageSettingsProvider \
-    Updater
-
-PRODUCT_COPY_FILES += \
-    vendor/lineage/prebuilt/common/etc/init/init.lineage-updater.rc:$(TARGET_COPY_OUT_SYSTEM_EXT)/etc/init/init.lineage-updater.rc
 
 # Config
 PRODUCT_PACKAGES += \
@@ -149,13 +143,9 @@ PRODUCT_PACKAGES += \
 
 # Extra tools in Lineage
 PRODUCT_PACKAGES += \
-    bash \
     curl \
     getcap \
-    htop \
-    nano \
     setcap \
-    vim
 
 PRODUCT_PACKAGES += \
     nano_recovery
@@ -178,23 +168,6 @@ PRODUCT_ARTIFACT_PATH_REQUIREMENT_ALLOWED_LIST += \
     system/%/libfuse-lite.so \
     system/%/libntfs-3g.so
 
-# Openssh
-PRODUCT_PACKAGES += \
-    scp \
-    sftp \
-    ssh \
-    sshd \
-    sshd_config \
-    ssh-keygen \
-    start-ssh
-
-PRODUCT_COPY_FILES += \
-    vendor/lineage/prebuilt/common/etc/init/init.openssh.rc:$(TARGET_COPY_OUT_PRODUCT)/etc/init/init.openssh.rc
-
-# rsync
-PRODUCT_PACKAGES += \
-    rsync
-
 # Storage manager
 PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
     ro.storage_manager.enabled=true
@@ -227,7 +200,7 @@ PRODUCT_DEXPREOPT_SPEED_APPS += \
     SystemUI
 
 PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
-    dalvik.vm.systemuicompilerfilter=speed
+    dalvik.vm.systemuicompilerfilter=everything
 
 ifeq ($(TARGET_BUILD_VARIANT),userdebug)
 PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
diff --git a/config/common_mobile.mk b/config/common_mobile.mk
index a09a9ef..4873940 100644
--- a/config/common_mobile.mk
+++ b/config/common_mobile.mk
@@ -49,37 +49,8 @@ endif
 
 # Customizations
 PRODUCT_PACKAGES += \
-    IconPackCircularAndroidOverlay \
-    IconPackCircularLauncherOverlay \
-    IconPackCircularSettingsOverlay \
-    IconPackCircularSystemUIOverlay \
-    IconPackFilledAndroidOverlay \
-    IconPackFilledLauncherOverlay \
-    IconPackFilledSettingsOverlay \
-    IconPackFilledSystemUIOverlay \
-    IconPackKaiAndroidOverlay \
-    IconPackKaiLauncherOverlay \
-    IconPackKaiSettingsOverlay \
-    IconPackKaiSystemUIOverlay \
-    IconPackRoundedAndroidOverlay \
-    IconPackRoundedLauncherOverlay \
-    IconPackRoundedSettingsOverlay \
-    IconPackRoundedSystemUIOverlay \
-    IconPackSamAndroidOverlay \
-    IconPackSamLauncherOverlay \
-    IconPackSamSettingsOverlay \
-    IconPackSamSystemUIOverlay \
-    IconPackVictorAndroidOverlay \
-    IconPackVictorLauncherOverlay \
-    IconPackVictorSettingsOverlay \
-    IconPackVictorSystemUIOverlay \
-    IconShapePebbleOverlay \
-    IconShapeRoundedRectOverlay \
-    IconShapeSquareOverlay \
-    IconShapeSquircleOverlay \
-    IconShapeTaperedRectOverlay \
-    IconShapeTeardropOverlay \
-    IconShapeVesselOverlay \
+    NavigationBarMode3ButtonOverlay \
+    NavigationBarModeGesturalOverlay \
     LineageNavigationBarNoHint
 
 # Legal
@@ -109,6 +80,4 @@ PRODUCT_ARTIFACT_PATH_REQUIREMENT_ALLOWED_LIST += \
 
 # Themes
 PRODUCT_PACKAGES += \
-    LineageBlackTheme \
     ThemePicker \
-    ThemesStub
diff --git a/config/common_mobile_full.mk b/config/common_mobile_full.mk
index 22b1acf..0aba249 100644
--- a/config/common_mobile_full.mk
+++ b/config/common_mobile_full.mk
@@ -3,17 +3,12 @@ $(call inherit-product, vendor/lineage/config/common_mobile.mk)
 
 PRODUCT_SIZE := full
 
-# Include {Lato,Rubik} fonts
-$(call inherit-product-if-exists, external/google-fonts/lato/fonts.mk)
-$(call inherit-product-if-exists, external/google-fonts/rubik/fonts.mk)
-
 # Apps
 PRODUCT_PACKAGES += \
     Camelot \
     Etar \
     Profiles \
     Recorder \
-    Seedvault \
     Twelve
 
 ifneq ($(PRODUCT_NO_CAMERA),true)
@@ -21,22 +16,11 @@ PRODUCT_PACKAGES += \
     Aperture
 endif
 
-ifneq ($(TARGET_EXCLUDES_AUDIOFX),true)
-PRODUCT_PACKAGES += \
-    AudioFX
-endif
-
 # Extra cmdline tools
 PRODUCT_PACKAGES += \
     unrar \
     zstd
 
-# Fonts
-PRODUCT_PACKAGES += \
-    fonts_customization.xml \
-    FontLatoOverlay \
-    FontRubikOverlay
-
 # Include Lineage LatinIME dictionaries
 PRODUCT_PACKAGE_OVERLAYS += vendor/lineage/overlay/dictionaries
 PRODUCT_ENFORCE_RRO_EXCLUDED_OVERLAYS += vendor/lineage/overlay/dictionaries
diff --git a/config/telephony.mk b/config/telephony.mk
index 0c6ebba..6878efb 100644
--- a/config/telephony.mk
+++ b/config/telephony.mk
@@ -14,7 +14,6 @@ PRODUCT_PACKAGES += \
 # Telephony packages
 PRODUCT_PACKAGES += \
     messaging \
-    Stk
 
 # Default ringtone
 PRODUCT_PRODUCT_PROPERTIES += \
